pennylane_lightning/core/_adjoint_jacobian_base.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Internal methods for adjoint Jacobian differentiation method.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any, Callable, List
+
+import numpy as np
+import pennylane as qml
+from pennylane import BasisState, QuantumFunctionError, StatePrep
+from pennylane.measurements import Expectation, MeasurementProcess, State
+from pennylane.operation import Operation
+from pennylane.tape import QuantumTape
+
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+
+
+class LightningBaseAdjointJacobian(ABC):
+    """Lightning [Device] Adjoint Jacobian class
+
+    A class that serves as a base class for Lightning state-vector simulators.
+    Check and execute the adjoint Jacobian differentiation method.
+
+    Args:
+        qubit_state(Lightning[Device]StateVector): State Vector to calculate the adjoint Jacobian with.
+        batch_obs(bool): If serialized tape is to be batched or not.
+    """
+
+    def __init__(self, qubit_state: Any, batch_obs: bool) -> None:
+        self._qubit_state = qubit_state
+        self._batch_obs = batch_obs
```

Comment:
@LuisAlfredoNu Could you add tests for the base class methods?

Reply:
We should have unit tests for all these methods. They don't need to be complicated, but ensure that you test the expected logic and corner cases.

Reply:
As we discussed offline, the creation of a test for the base class will be postponed until the migration of LGPU to the new device API.